src/controllers/filters/parser/Tokenizer.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "controllers/filters/parser/Types.hpp"
+
+namespace filterparser {
+
+static const QStringList validIdentifiers = {
+    "author.badges",          // String list
+    "author.color",           // QColor
+    "author.name",            // String
+    "author.no_color",        // Bool
+    "author.subbed",          // Bool
+    "author.sub_length",      // Int
+    "channel.name",           // String
+    "channel.watching",       // Bool
+    "flags.highlighted",      // Bool
+    "flags.points_redeemed",  // Bool
+    "flags.sub_message",      // Bool
+    "flags.system_message",   // Bool
+    "flags.whisper",          // Bool
+    "message.content",        // String
+    "message.length"          // Int
+};
+
+static const QStringList friendlyValidIdentifiers = {
+    "author badges",        "author color",       "author name",
+    "author has no color?", "author subscribed?", "author sub length",
+    "channel name",         "/watching channel?", "highlighted?",
+    "redeemed points?",     "sub/resub message?", "whisper message?",
+    "message text",         "message length",
+};
```

Comment:
these two QStringLists are not the same length. let's try to prevent that from happening again by building a QMap<QString, QString> where the key is the identifier and the value is the identifier friendly name, so you can just do validIdentifiers = map.keys(); friendlyValidIdentifiers = map.values();

it's "system message" missing btw

Reply:
fixed in 1da48d1 👍